File: 242_Valid_Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size()!=t.size())
        return false;

        unordered_map<char,int> m;

        for(auto x: s){
            m[x]++;
        }

        for(auto x:t){
            m[x]--;
        }

        for(auto x:m){
            if(x.second!=0)
            return false;
        }

        return true;

        
    }
};

File: 242_Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        # sorted_s=sorted(s)
        # sorted_t=sorted(t)

        # return sorted_s==sorted_t

        count=defaultdict(int) #default make count as 0

        for x in s:
            count[x]+=1
        
        for x in t:
            count[x]-=1
        
        for val in count.values():
            if val!=0:
                return False
        

        return True


        
